src/java.desktop/share/classes/javax/swing/JTable.java

```diff
@@ -3006,14 +3006,11 @@ else if (column >= getColumnCount()) {
         }
         else {
             TableColumnModel cm = getColumnModel();
-            if( getComponentOrientation().isLeftToRight() ) {
-                for(int i = 0; i < column; i++) {
-                    r.x += cm.getColumn(i).getWidth();
-                }
-            } else {
-                for(int i = cm.getColumnCount()-1; i > column; i--) {
-                    r.x += cm.getColumn(i).getWidth();
-                }
+            for(int i = 0; i < column; i++) {
+                r.x += cm.getColumn(i).getWidth();
+            }
+            if( !getComponentOrientation().isLeftToRight() ) {
+                r.x = getWidthInRightToLeft() - r.x - cm.getColumn(column).getWidth();
```

Comment:
It will be good to add comments explaining why you are doing what you are doing for future recall and
more is less here...